server/src/jobs/assets.h

```diff
@@ -16,27 +16,42 @@
 
 #pragma once
 #include "discovery-task.h"
+#include <set>
 
 // =====================================================================================================================
 
 namespace fty::job {
 
+struct DeviceInfo {
+    int indexDevice;
+    std::string& deviceSerial;
+    std::string type;
+};
+
 /// Discover Assets from enpoint
 /// Returns @ref commands::assets::Out (something like a list of Assets)
 class Assets : public Task<Assets, commands::assets::In, commands::assets::Out>
 {
 public:
     using Task::Task;
 
+    // Get assets list
+    Expected<void> getAssets(const commands::assets::In& in, commands::assets::Out& out);
+
+    // Add asset sensors
+    void addSensors(const DeviceInfo& deviceInfo, int& indexSensor, pack::ObjectList<fty::commands::assets::Asset>& sensors, const std::map<std::string, std::string>& dump);
```

Comment:
To be honest, I don't like this function notation, you have here (in, out, out, in)
Maybe better you will return all out as tuple?